src/riak_core_coverage_plan.erl

```diff
@@ -177,13 +180,22 @@ replace_traditional_chunk(VnodeIdx, Node, Filters, NVal,
 %%      originally designed for parallel extraction of data via 2i.
 -spec create_subpartition_plan('all'|'allup',
                                pos_integer(),
-                               pos_integer(),
+                               pos_integer() | {pos_integer(), pos_integer()},
                                pos_integer(),
                                req_id(), atom()) ->
                                       {error, term()} | subp_plan().
+create_subpartition_plan(VNodeTarget, NVal, {MinPar, RingSize}, PVC, ReqId, Service) ->
+    Count =
+        if
+            MinPar =< RingSize ->
+                RingSize;
+            true ->
+                next_power_of_two(MinPar)
+        end,
+    create_subpartition_plan(VNodeTarget, NVal, Count, PVC, ReqId, Service);
 create_subpartition_plan(VNodeTarget, NVal, Count, PVC, ReqId, Service) ->
-    %% IMPORTANT: `Count' is assumed to be a power of 2 as determined
-    %% by `riak_client'. Anything else will behave badly.
+    %% IMPORTANT: `Count' is assumed to be a power of 2. Anything else
```

Comment:
I would do a check on the power of 2 and crash here instead of carrying on with something we now will behave badly.